test/AutoDiff/differentiable_attr_type_checking.swift

```diff
@@ -846,17 +876,7 @@ func inout2(x: Float, y: inout Float) -> Float {
   let _ = x + y
 }
 
-
-// Missing `@differentiable` attribute, without printing the 'wrt' arguments.
-
-protocol DifferentiableWhereClause: Differentiable {
```

Comment:
Note: reordered tests.
- Renamed `DifferentiableWhereClause` to `ProtocolRequirementUnsupported` below.
- Moved `MissingDifferentiableTest` to `ProtocolRequirementsRefined` above.